ddl/partition.go

```diff
@@ -702,6 +702,9 @@ func checkRangePartitioningKeysConstraints(sctx sessionctx.Context, s *ast.Creat
 		partCols = columnInfoSlice(partColumns)
 	} else if len(s.Partition.ColumnNames) > 0 {
 		partCols = columnNameSlice(s.Partition.ColumnNames)
+	} else {
+		// TODO: Check keys constraints for list, key partition type and so on.
+		return nil
```

Comment:
should we return an error instead?

Reply:
I doubt that if the TODO will be addressed in the future, maybe a warning will force us to fix it in the future?

Reply:
When we implement key partition one day, this TODO **must** be addressed. @wjhuang2016 
Do not return error because the user can still user `create table partition by key ...`, though the table is treated as normal table instead of partition table.
Error is not user-friendly. @jackysp

Reply:
I thought that when we turned on the partition table feature, we would return errors for unsupported partition types. When the user enables the partition table feature, but creates a partition table and finds that it is not actually a partition table, and there are no errors/warnings reported. Users may be confused.

Reply:
This is all due to our past relaxed checking. Perhaps a strictly checked configuration should be opened for the partition table in the future.